Microsoft.Toolkit.Win32/Microsoft.Toolkit.Win32.UI.Controls/Interop/WinRT/WebViewControlProcessOptions.cs

```diff
@@ -13,45 +14,91 @@ namespace Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT
 
+        /// <summary>
+        /// Gets or sets the partition for the web view.
+        /// </summary>
+        /// <value>The partition.</value>
+        public string Partition { get; set; }
+
         /// <summary>
         /// Gets or sets the private network client server capability.
         /// </summary>
         /// <value>The private network client server capability.</value>
         public WebViewControlProcessCapabilityState PrivateNetworkClientServerCapability { get; set; }
 
+        /// <summary>
+        /// Gets or sets the user agent.
+        /// </summary>
+        /// <value>The user agent.</value>
+        public string UserAgent { get; set; }
+
         /// <summary>
         /// Initializes a new instance of the <see cref="WebViewControlProcessOptions"/> class.
         /// </summary>
         public WebViewControlProcessOptions()
         {
             EnterpriseId = string.Empty;
+            Partition = string.Empty;
+            UserAgent = string.Empty;
             PrivateNetworkClientServerCapability = WebViewControlProcessCapabilityState.Default;
         }
 
         /// <summary>
-        /// Converts this instance to a <seealso cref="Windows.Web.UI.Interop.WebViewControlProcessOptions"/> instance.
+        /// Performs an implicit conversion from <see cref="Windows.Web.UI.Interop.WebViewControlProcessOptions"/> to <see cref="WebViewControlProcessOptions"/>.
         /// </summary>
-        /// <returns>A <seealso cref="Windows.Web.UI.Interop.WebViewControlProcessOptions"/> instance.</returns>
-        internal Windows.Web.UI.Interop.WebViewControlProcessOptions ToWinRtWebViewControlProcessOptions()
+        /// <param name="options">The options.</param>
+        /// <returns>The result of the conversion.</returns>
+        public static implicit operator WebViewControlProcessOptions(Windows.Web.UI.Interop.WebViewControlProcessOptions options) => ToWinRtWebViewControlProcessOptions(options);
+
+        public static Windows.Web.UI.Interop.WebViewControlProcessOptions ToWinRtWebViewControlProcessOptions(WebViewControlProcessOptions options)
         {
             var retval = new Windows.Web.UI.Interop.WebViewControlProcessOptions();
 
-            if (!string.IsNullOrEmpty(EnterpriseId) && !StringComparer.InvariantCulture.Equals(retval.EnterpriseId, EnterpriseId))
+            if (!string.IsNullOrEmpty(options?.EnterpriseId) && !StringComparer.InvariantCulture.Equals(retval.EnterpriseId, options?.EnterpriseId))
+            {
+                retval.EnterpriseId = options.EnterpriseId;
+            }
+
+            if (ApiInformation.IsPropertyPresent(
+                   "Windows.Web.UI.Interop.WebViewControlProcessOptions",
+                   "Partition"))
             {
-                retval.EnterpriseId = EnterpriseId;
+                if (!string.IsNullOrEmpty(options?.Partition))
+                {
+                    retval.Partition = options?.Partition;
```

Comment:
This is safe now as it's checked so '?' is redundant? (same below)

Reply:
Fixed in 0091a1fb1161819e9ebaba84344d18695ea3f134